### PR TITLE
Added unit tests

### PR DESCRIPTION
**Pull request by yanayukhimuk**

### CODE REVIEW

File: BLL/Services/EventService.cs [hunk 1]
```diff
@@ -15,26 +16,26 @@ public class EventService(IEventRepository eventRepository, ISeatRepository seat
...
+        public async Task<IEnumerable<Event>> ListAllEventsAsync()
         {
             return await _eventRepository.GetAllAsync();
         }
 
-        public async Task<IEnumerable<Seat>> ListAllSeatsAndPricesByEventIdAsync(Guid event_id, Guid section_id)
+        public async Task<IEnumerable<SeatInfo>> ListAllSeatsAndPricesByEventIdAsync(Guid event_id, Guid section_id)
         {
             var seats = await _seatRepository.GetAllAsync();
             var reqSeats = seats.Where(s => s.EventId == event_id && s.SectionId == section_id)
```
Comment (PavelHanenia) [inline BLL/Services/EventService.cs:27]: This is an example of a bad query. You get all the tickets and then filter them in memory. You should better have some method in your _seatRepository like this:
```
public async Task<ICollection<Seat>> GetByEventIdSectionId(Guid eventId, Guid sectionId)
{
    return await _dbSet.Where(seat => seat.EventId == eventId && seat.SectionId == sectionId).ToListAsync();
}
```
Then all filtering will be on the DB side. The same for other calls.

File: DAL/Repositories/GenericRepository.cs [hunk 2]
```diff
@@ -10,9 +11,9 @@ public class GenericRepository<T>(TicketBookingSystemDbContext context) where T
         protected readonly TicketBookingSystemDbContext _context = context;
         protected readonly DbSet<T> _dbSet = context.Set<T>();
 
-        public IEnumerable<T> GetAll()
+        public IQueryable<T> GetAll()
```
Comment (PavelHanenia) [inline DAL/Repositories/GenericRepository.cs:14]: Why do we need to return IQueriable from the repository? Here you can read why you should not do this:
https://stackoverflow.com/questions/33755499/entity-framework-repository-pattern-why-not-return-iqueryable
https://weblogs.asp.net/dotnetstories/repository-iqueryable

File: TicketBookingSystemCore/Controllers/OrderController.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public async Task<IActionResult> GetCartItems(Guid cart_id)
         {
             var seats = await _orderService.GetCartItems(cart_id);
 
-            if (seats == null)
+            if (seats == null || !seats.Any())
             {
                 return NotFound();
             }
```
Comment (PavelHanenia) [inline TicketBookingSystemCore/Controllers/OrderController.cs:21]: This validation looks odd - you check seats after getting them from service. In this case - `Get By CartId` we should rather fail inside service with bad data and throw an exception if CartId or items for CartId are missing or other validations failed, and then handle particular exceptions in ExceptionFilter returning the appropriate response status code.

File: TicketBookingSystemCore/Controllers/OrderController.cs [hunk 4]
```diff
@@ -27,20 +27,41 @@ public async Task<IActionResult> GetCartItems(Guid cart_id)
...
         [HttpDelete("carts/{cart_id}/events/{event_id}/seats/{seat_id}")]
         public async Task<IActionResult> DeleteSeatFromCart(Guid cart_id, Guid event_id, Guid seat_id)
         {
             await _orderService.DeleteSeatFromCart(cart_id, event_id, seat_id);
+            var seats = await _orderService.GetCartItems(cart_id);
+
+            if (seats.Any(s => s.SeatId == seat_id))
+            {
+                return BadRequest();
+            }
```
Comment (PavelHanenia) [inline TicketBookingSystemCore/Controllers/OrderController.cs:49]: Same here and in other endpoints.

File: TicketBookingSystemCore/Controllers/OrderController.cs [hunk 5]
```diff
@@ -27,20 +27,41 @@ public async Task<IActionResult> GetCartItems(Guid cart_id)
         public async Task<IActionResult> AddSeatToCart(Guid cart_id, [FromBody] Seat seat)
         {
             await _orderService.AddSeatToCart(cart_id, seat);
+            var seats = await _orderService.GetCartItems(cart_id);
+
+            if (seats == null || seats.Any(s => s.SeatId != seat.SeatId))
+            {
+                return NotFound();
+            }
```
Comment (PavelHanenia) [inline TicketBookingSystemCore/Controllers/OrderController.cs:35]: Same here.

File: TicketBookingSystemUnitTests/UnitTests/EventControllerTests.cs [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+using NuGet.Protocol.Core.Types;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Design;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TicketBookingSystemCore.Controllers;
+using TicketBookingSystemUnitTests.Mocks;
+using TicketBookingSystemUnitTests.TestData;
```
Comment (PavelHanenia) [inline TicketBookingSystemUnitTests/UnitTests/EventControllerTests.cs:21]: Do we need all these usings? please, check all files.

File: TicketBookingSystemUnitTests/UnitTests/OrdersControllerTests.cs [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+            using (new AssertionScope())
+            {
+                result.Should().BeOfType<NotFoundResult>("The cart does not exist");
+            }
+        }
+
+        [Fact]
+        public async Task AddSeatToCart_ReturnsNotFound_WhenSeatNotAdded()
+        {
```
Comment (PavelHanenia) [inline TicketBookingSystemUnitTests/UnitTests/OrdersControllerTests.cs:122]: This test fails for me:
image

File: DAL/Repositories/GenericRepository.cs (hunk 2 above)
Comment (yanayukhimuk) [reply DAL/Repositories/GenericRepository.cs:14]: Fixed

File: TicketBookingSystemCore/Controllers/OrderController.cs (hunk 3 above)
Comment (yanayukhimuk) [reply TicketBookingSystemCore/Controllers/OrderController.cs:21]: Removed validations from controllers

File: TicketBookingSystemCore/Controllers/OrderController.cs (hunk 4 above)
Comment (yanayukhimuk) [reply TicketBookingSystemCore/Controllers/OrderController.cs]: Removed validations from controllers

File: TicketBookingSystemCore/Controllers/OrderController.cs (hunk 5 above)
Comment (yanayukhimuk) [reply TicketBookingSystemCore/Controllers/OrderController.cs]: Removed validations from controllers

File: TicketBookingSystemUnitTests/UnitTests/EventControllerTests.cs (hunk 6 above)
Comment (yanayukhimuk) [reply TicketBookingSystemUnitTests/UnitTests/EventControllerTests.cs]: Removed usings

File: TicketBookingSystemUnitTests/UnitTests/OrdersControllerTests.cs (hunk 7 above)
Comment (yanayukhimuk) [reply TicketBookingSystemUnitTests/UnitTests/OrdersControllerTests.cs:140]: Tried to fix - do not know how to mock throwing exception

**Pull request merged**